src/cpu/matmul/cpu_matmul_list.cpp

```diff
@@ -78,8 +78,8 @@ constexpr impl_list_item_t impl_list[] = REG_MATMUL_P({
         CPU_INSTANCE_AVX2(brgemm_matmul_t<avx2_vnni_2>)
         CPU_INSTANCE_AVX2(brgemm_matmul_t<avx2_vnni>)
         CPU_INSTANCE_AVX2(brgemm_matmul_t<avx2>)
-        CPU_INSTANCE_RV64GCV(rvv_matmul_t)
         CPU_INSTANCE(gemm_f32_matmul_t)
+        CPU_INSTANCE_RV64GCV(rvv_matmul_t)
```

Comment:
Why would you want to move HW-specific implementation behind a generic f32 C-based one?